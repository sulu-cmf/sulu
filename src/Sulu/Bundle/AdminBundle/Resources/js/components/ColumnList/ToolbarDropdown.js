// @flow
import React, {Fragment} from 'react';
import type {ElementRef} from 'react';
import classNames from 'classnames';
import {observable, action} from 'mobx';
import {observer} from 'mobx-react';
import Icon from '../Icon';
import Popover from '../Popover';
import ToolbarDropdownList from './ToolbarDropdownList';
import type {ToolbarDropdown as ToolbarDropdownProps} from './types';
import toolbarStyles from './toolbar.scss';

@observer
export default class ToolbarDropdown extends React.Component<ToolbarDropdownProps> {
    static defaultProps = {
        skin: 'primary',
    };

    @observable popoverOpen: boolean = false;
    @observable popoverAnchorElement: ?ElementRef<*>;

    @action handleClick = (event: SyntheticEvent<HTMLOptionElement>) => {
        this.popoverAnchorElement = event.currentTarget;
        this.popoverOpen = true;
    };

    @action handlePopoverClose = () => {
        this.popoverOpen = false;
    };

    render() {
        const {icon, options, skin, columnIndex} = this.props;

        const className = classNames(
            toolbarStyles.item,
            toolbarStyles[skin]
        );

        return (
            <Fragment>
                <div onClick={this.handleClick} className={className}>
                    <Icon name={icon} />
                </div>
                <Popover
                    open={this.popoverOpen}
                    anchorElement={this.popoverAnchorElement}
                    onClose={this.handlePopoverClose}
                >
                    {
                        (setPopoverElementRef, popoverStyle) => (
                            <div
                                style={popoverStyle}
                                ref={setPopoverElementRef}
                            >
                                <ToolbarDropdownList
                                    columnIndex={columnIndex}
                                    onClick={this.handlePopoverClose}
                                    options={options}
                                />
                            </div>
                        )
                    }
                </Popover>
            </Fragment>
        );
    }
}

