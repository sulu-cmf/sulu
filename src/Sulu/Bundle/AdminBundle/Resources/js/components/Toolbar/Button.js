// @flow
import classNames from 'classnames';
import React from 'react';
import type {ElementRef} from 'react';
import Icon from '../Icon';
import Loader from '../Loader';
import type {Button as ButtonProps} from './types';
import buttonStyles from './button.scss';

const LOADER_SIZE = 20;
const ICON_ANGLE_DOWN = 'su-angle-down';

export default class Button extends React.PureComponent<ButtonProps> {
    static defaultProps = {
        active: false,
        disabled: false,
        hasOptions: false,
        primary: false,
        showText: true,
    };

    handleOnClick = () => {
        this.props.onClick();
    };

    setButtonRef = (ref: ?ElementRef<'button'>) => {
        const {buttonRef} = this.props;
        if (buttonRef && ref) {
            buttonRef(ref);
        }
    };

    render() {
        const {
            icon,
            size,
            skin,
            value,
            active,
            primary,
            loading,
            disabled,
            hasOptions,
            showText,
        } = this.props;
        const buttonClass = classNames(
            buttonStyles.button,
            {
                [buttonStyles.active]: active,
                [buttonStyles[size]]: size,
                [buttonStyles[skin]]: skin,
                [buttonStyles.primary]: primary,
            }
        );
        const buttonContent = this.props.children || value;

        return (
            <button
                className={buttonClass}
                disabled={disabled}
                onClick={this.handleOnClick}
                ref={this.setButtonRef}
                value={value}
            >
                {loading &&
                    <Loader className={buttonStyles.loader} size={LOADER_SIZE} />
                }
                {icon &&
                    <Icon className={buttonStyles.icon} name={icon} />
                }
                {(buttonContent && showText) &&
                    <span className={buttonStyles.label}>{buttonContent}</span>
                }
                {hasOptions &&
                    <Icon className={buttonStyles.dropdownIcon} name={ICON_ANGLE_DOWN} />
                }
            </button>
        );
    }
}
