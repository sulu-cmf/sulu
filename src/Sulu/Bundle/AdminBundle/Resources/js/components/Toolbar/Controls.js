// @flow
import type {ChildrenArray} from 'react';
import React from 'react';
import classNames from 'classnames';
import type {Group, Item, Skin} from './types';
import controlsStyles from './controls.scss';

type Props = {
    children: ChildrenArray<Item | Group | false>,
    grow?: boolean,
    skin?: Skin,
};

export default class Controls extends React.PureComponent<Props> {
    static defaultProps = {
        grow: false,
        skin: 'light',
    };

    static createChildren(children: ChildrenArray<Item | Group | false>, skin?: Skin) {
        return React.Children.map(children, (child: Item | Group | false) => {
            if (!child) {
                return;
            }

            // $FlowFixMe
            return React.cloneElement(
                child,
                {
                    ...child.props,
                    skin: skin,
                }
            );
        });
    }

    render() {
        const {
            children,
            grow,
            skin,
        } = this.props;

        const controlsClass = classNames(
            controlsStyles.controls,
            controlsStyles[skin],
            {
                [controlsStyles.grow]: grow,
            }
        );

        return (
            <div className={controlsClass}>
                {Controls.createChildren(children, skin)}
            </div>
        );
    }
}
