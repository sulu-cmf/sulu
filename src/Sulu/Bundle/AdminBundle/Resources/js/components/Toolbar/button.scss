@import '../../containers/Application/colors.scss';

$buttonTextLightColor: $blueZodiac;
$buttonTextDarkColor: $white;
$buttonTextLightColorActive: $shakespeare;
$buttonTextDarkColorActive: $white;
$buttonTextLightColorDisabled: $silver;
$buttonTextDarkColorDisabled: $stormGray;
$buttonActiveBackgroundLightColor: $wildSand;
$buttonActiveBackgroundDarkColor: $mirage;
$primaryTextColor: $white;
$primaryBackgroundColor: $shakespeare;

.light {
    color: $buttonTextLightColor;

    &:disabled {
        color: $buttonTextLightColorDisabled;
    }

    &:not(:disabled) {
        &:hover,
        &.active {
            color: $buttonTextLightColorActive;
            background-color: $buttonActiveBackgroundLightColor;
        }
    }
}

.dark {
    color: $buttonTextDarkColor;

    &:disabled {
        color: $buttonTextDarkColorDisabled;
    }

    &:not(:disabled) {
        &:hover,
        &.active {
            color: $buttonTextDarkColorActive;
            background-color: $buttonActiveBackgroundDarkColor;
        }
    }
}

.button {
    display: block;
    min-width: 60px;
    position: relative;
    padding: 0 20px;
    border: none;
    height: 60px;
    background-color: transparent;
    font-size: 12px;
    cursor: pointer;

    &:disabled {
        cursor: default;
    }

    &:not(:disabled) {
        &.active {
            z-index: 2;
        }
    }

    &.small {
        padding: 0;

        .dropdown-icon {
            position: absolute;
            left: 50%;
            bottom: 8px;
            margin-left: -6px;
            padding: 0;
        }
    }

    &.primary,
    &.primary:hover {
        color: $primaryTextColor;
        background-color: $primaryBackgroundColor;
    }
}

.icon {
    vertical-align: middle;
    font-size: 20px;

    & + .label {
        padding: 0 0 0 10px;
    }

    .loader ~ & {
        visibility: hidden;
    }
}

.label {
    vertical-align: middle;

    .loader ~ & {
        visibility: hidden;
    }
}

.dropdown-icon {
    padding: 0 0 0 10px;

    .loader ~ & {
        visibility: hidden;
    }
}

.loader {
    position: absolute;
    top: 20px;
    left: 0;
    right: 0;
}
