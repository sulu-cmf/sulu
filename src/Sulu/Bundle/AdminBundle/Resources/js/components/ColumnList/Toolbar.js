// @flow
import React from 'react';
import type {ElementRef} from 'react';
import {action, observable} from 'mobx';
import {observer} from 'mobx-react';
import ToolbarDropdown from './ToolbarDropdown';
import ToolbarButton from './ToolbarButton';
import type {ToolbarItemConfig} from './types';
import toolbarStyles from './toolbar.scss';

type Props = {|
    columnIndex: number,
    toolbarItems: Array<ToolbarItemConfig>,
    toolbarRef?: (?ElementRef<'div'>) => void,
|};

@observer
export default class Toolbar extends React.Component<Props> {
    static defaultProps = {
        toolbarItems: [],
    };

    @observable toolbar: ElementRef<'div'>;

    @action setToolbarRef = (ref: ?ElementRef<'div'>) => {
        const {toolbarRef} = this.props;

        if (toolbarRef) {
            toolbarRef(ref);
        }
    };

    renderToolbarItems = (toolbarItems: Array<ToolbarItemConfig>) => {
        return toolbarItems.map((toolbarItemConfig: ToolbarItemConfig, index: number) => {
            switch (toolbarItemConfig.type) {
                case 'dropdown':
                    return <ToolbarDropdown key={index} columnIndex={this.props.columnIndex} {...toolbarItemConfig} />;
                case 'button':
                    return <ToolbarButton key={index} columnIndex={this.props.columnIndex} {...toolbarItemConfig} />;
                default:
                    throw new Error('Unknown toolbar item type given: "' + toolbarItemConfig.type + '"');
            }
        });
    };

    render() {
        const {toolbarItems} = this.props;

        return (
            <div
                ref={this.setToolbarRef}
                className={toolbarStyles.toolbar}
            >
                {this.renderToolbarItems(toolbarItems)}
            </div>
        );
    }
}

