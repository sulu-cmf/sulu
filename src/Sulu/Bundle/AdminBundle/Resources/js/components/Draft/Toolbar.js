// @flow
import React from 'react';
import {EditorState, RichUtils, Modifier} from 'draft-js';
import Button from './Button';
import Spacer from './Spacer';
import toolbarStyles from './toolbar.scss';
import type {ToolbarItem, ToolbarConfig} from './types';

type Props = {
    editorState: EditorState,
    onChange: (state: EditorState) => any,
};

// TODO: make configurable
const toolbarConfig: ToolbarConfig = [
    {type: 'button', handle: 'inline-style', style: 'BOLD', icon: 'su-bold'},
    {type: 'button', handle: 'inline-style', style: 'ITALIC', icon: 'su-italic'},
    {type: 'button', handle: 'inline-style', style: 'UNDERLINE', icon: 'su-underline'},
    {type: 'button', handle: 'inline-style', style: 'STRIKETHROUGH', icon: 'su-strikethrough'},
    {type: 'button', handle: 'inline-style', style: 'SUPERSCRIPT', icon: 'su-superscript'},
    {type: 'button', handle: 'inline-style', style: 'SUBSCRIPT', icon: 'su-subscript'},
    {type: 'spacer'},
    {type: 'button', handle: 'block-type', style: 'ordered-list-item', icon: 'su-list-ol'},
    {type: 'button', handle: 'block-type', style: 'unordered-list-item', icon: 'su-list-ul'},
    {type: 'spacer'},
    {type: 'button', handle: 'alignment', style: 'left', icon: 'su-align-left'},
    {type: 'button', handle: 'alignment', style: 'center', icon: 'su-align-center'},
    {type: 'button', handle: 'alignment', style: 'right', icon: 'su-align-right'},
    {type: 'button', handle: 'alignment', style: 'justify', icon: 'su-align-justify'},
    {type: 'spacer'},
    {type: 'button', handle: 'anchor', style: 'anchor', icon: 'su-link2'},
];

const ToolbarItemTypes = {
    Button: 'button',
    Spacer: 'spacer',
};

export default class Toolbar extends React.Component<Props> {
    handleToggle = (handle: string, style: string) => {
        switch (handle) {
            case 'inline-style':
                this.handleToggleInlineStyle(style);
                break;
            case 'block-type':
                this.handleToggleBlockType(style);
                break;
            case 'alignment':
                this.handleToggleAlignment(style);
                break;
            case 'anchor':
                this.handleAnchor();
                break;
        }
    };

    handleToggleInlineStyle = (inlineStyle: string) => {
        let newEditorState = RichUtils.toggleInlineStyle(
            this.props.editorState,
            inlineStyle
        );

        if (inlineStyle === 'SUPERSCRIPT' || inlineStyle === 'SUBSCRIPT') {
            const removeStyle = inlineStyle === 'SUBSCRIPT' ? 'SUPERSCRIPT' : 'SUBSCRIPT';
            const contentState = Modifier.removeInlineStyle(
                newEditorState.getCurrentContent(),
                newEditorState.getSelection(),
                removeStyle
            );
            newEditorState = EditorState.push(newEditorState, contentState, 'change-inline-style');
        }

        if (newEditorState) {
            this.props.onChange(newEditorState);
        }
    };

    handleToggleBlockType = (blockType: string) => {
        this.props.onChange(
            RichUtils.toggleBlockType(
                this.props.editorState,
                // $FlowFixMe
                blockType
            )
        );
    };

    handleToggleAlignment = (alignment: string) => {
        const {
            editorState,
        } = this.props;

        const contentState = Modifier.setBlockData(
            editorState.getCurrentContent(),
            editorState.getSelection(),
            { 'text-align': alignment });

        const newEditorState = EditorState.push(editorState, contentState, 'change-block-data');

        if (newEditorState) {
            this.props.onChange(newEditorState);
        }
    };

    handleAnchor = () => {
        const {editorState} = this.props;
        const selection = editorState.getSelection();
        if (!selection.isCollapsed()) {
            const contentState = editorState.getCurrentContent();
            const startKey = editorState.getSelection().getStartKey();
            const startOffset = editorState.getSelection().getStartOffset();
            const blockWithLinkAtBeginning = contentState.getBlockForKey(startKey);
            const linkKey = blockWithLinkAtBeginning.getEntityAt(startOffset);

            let url = '';
            if (linkKey) {
                const linkInstance = contentState.getEntity(linkKey);
                url = linkInstance.getData().url;
            }

            this.setState({
                showURLInput: true,
                urlValue: url,
            }, () => { });
        }
    };

    render() {
        const {editorState} = this.props;
        const currentInlineStyle = editorState.getCurrentInlineStyle();
        const selection = editorState.getSelection();
        const currentBlock = editorState.getCurrentContent().getBlockForKey(selection.getStartKey());
        const currentBlockType = currentBlock.getType();
        const currentAlignment = currentBlock.getData().get('text-align') || 'left';

        return (
            <div className={toolbarStyles.toolbar}>
                {
                    toolbarConfig.map(
                        (itemConfig: ToolbarItem, index: number) => {
                            switch (itemConfig.type) {
                                case ToolbarItemTypes.Button:
                                    let active = false;
                                    switch (itemConfig.handle) {
                                        case 'inline-style':
                                            active = currentInlineStyle.has(itemConfig.style);
                                            break;
                                        case 'block-type':
                                            active = itemConfig.style === currentBlockType;
                                            break;
                                        case 'alignment':
                                            active = itemConfig.style === currentAlignment;
                                    }

                                    return (
                                        <Button
                                            {...itemConfig}
                                            onToggle={this.handleToggle}
                                            active={active}
                                            key={index}
                                        />
                                    );
                                case ToolbarItemTypes.Spacer:
                                    return (<Spacer {...itemConfig} key={index} />);
                            }
                        }
                    )
                }
            </div>
        );
    }
}
