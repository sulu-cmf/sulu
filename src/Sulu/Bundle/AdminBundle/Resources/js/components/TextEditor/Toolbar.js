// @flow
import React from 'react';
import Button from './Button';
import ButtonGroup from './ButtonGroup';
import toolbarStyles from './toolbar.scss';

type Props = {
    value: Object,
    onChange: (change: Object) => void,
};

type ButtonConfig = {
    type: string,
    icon: string,
};

const DEFAULT_NODE = 'paragraph';

const toolbarConfig = {
    MARK_TYPE_BUTTONS: [
        {type: 'bold', icon: 'fa-bold'},
        {type: 'italic', icon: 'fa-italic'},
        {type: 'underline', icon: 'fa-underline'},
        {type: 'strikethrough', icon: 'fa-strikethrough'},
        {type: 'superscript', icon: 'fa-superscript'},
        {type: 'subscript', icon: 'fa-subscript'},
    ],
    BLOCK_TYPE_BUTTONS: [
        {type: 'ordered-list', icon: 'fa-list-ol'},
        {type: 'unordered-list', icon: 'fa-list-ul'},
    ],
};

export default class Toolbar extends React.Component<Props> {
    handleMarkType = (type: string) => {
        const { value } = this.props;
        this.props.onChange(
            value.change().toggleMark(type)
        );
    };

    handleBlockType = (type: string) => {
        const { value } = this.props;
        const change = value.change();
        const { document } = value;

        // Handle everything but list buttons.
        if (type != 'unordered-list' && type != 'ordered-list') {
            const isActive = this.hasBlock(type);
            const isList = this.hasBlock('list-item');

            if (isList) {
                change.setBlocks(isActive ? DEFAULT_NODE : type)
                    .unwrapBlock('unordered-list')
                    .unwrapBlock('ordered-list');
            } else {
                change.setBlocks(isActive ? DEFAULT_NODE : type);
            }
        } else {
            // Handle the extra wrapping required for list buttons.
            const isList = this.hasBlock('list-item');
            const isType = value.blocks.some((block) => {
                return !!document.getClosest(block.key, (parent) => parent.type == type);
            });

            if (isList && isType) {
                change.setBlocks(DEFAULT_NODE)
                    .unwrapBlock('unordered-list')
                    .unwrapBlock('ordered-list');
            } else if (isList) {
                change.unwrapBlock(type == 'unordered-list' ? 'ordered-list' : 'unordered-list')
                    .wrapBlock(type);
            } else {
                change.setBlocks('list-item').wrapBlock(type);
            }
        }

        this.props.onChange(change);
    };

    hasMark = (type: string) => {
        const {value} = this.props;
        return value.activeMarks.some((mark) => mark.type == type);
    };

    hasBlock = (type: string) => {
        const {value} = this.props;
        return value.blocks.some((node) => node.type == type);
    };

    render() {
        return (
            <div className={toolbarStyles.toolbar}>
                <ButtonGroup>
                    {
                        toolbarConfig.MARK_TYPE_BUTTONS.map((button: ButtonConfig, index: number) => (
                            <Button
                                key={index}
                                active={this.hasMark(button.type)}
                                type={button.type}
                                icon={button.icon}
                                onToggle={this.handleMarkType}
                            />
                        ))
                    }
                </ButtonGroup>
                <ButtonGroup>
                    {
                        toolbarConfig.BLOCK_TYPE_BUTTONS.map((button: ButtonConfig, index: number) => (
                            <Button
                                key={index}
                                active={this.hasBlock(button.type)}
                                type={button.type}
                                icon={button.icon}
                                onToggle={this.handleBlockType}
                            />
                        ))
                    }
                </ButtonGroup>
            </div>
        );
    }
}
