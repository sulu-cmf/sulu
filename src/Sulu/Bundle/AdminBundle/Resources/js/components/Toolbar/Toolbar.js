// @flow
import type {ChildrenArray, Element} from 'react';
import React from 'react';
import classNames from 'classnames';
import Button from './Button';
import Controls from './Controls';
import Dropdown from './Dropdown';
import Items from './Items';
import Icons from './Icons';
import Select from './Select';
import type {Skin} from './types';
import toolbarStyles from './toolbar.scss';

type Props = {
    children: ChildrenArray<Element<typeof Controls>>,
    skin?: Skin,
};

export default class Toolbar extends React.PureComponent<Props> {
    static defaultProps = {
        skin: 'light',
    };

    static Button = Button;

    static Controls = Controls;

    static Dropdown = Dropdown;

    static Items = Items;

    static Icons = Icons;

    static Select = Select;

    static createChildren(children: ChildrenArray<Element<typeof Controls>>, skin?: Skin) {
        return React.Children.map(children, (child) => {
            return React.cloneElement(
                child,
                {
                    ...child.props,
                    skin: skin,
                }
            );
        });
    }

    render() {
        const {
            skin,
            children,
        } = this.props;

        const toolbarClass = classNames(
            toolbarStyles.toolbar,
            toolbarStyles[skin]
        );

        return (
            <nav className={toolbarClass}>
                {Toolbar.createChildren(children, skin)}
            </nav>
        );
    }
}
