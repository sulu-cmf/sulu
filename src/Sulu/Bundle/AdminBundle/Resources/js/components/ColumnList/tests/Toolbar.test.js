// @flow
import React from 'react';
import {mount} from 'enzyme';
import pretty from 'pretty';
import Toolbar from '../Toolbar';
import ToolbarDropdown from '../ToolbarDropdown';

test('The Toolbar component should render with active', () => {
    const body = document.body;
    if (!body) {
        throw new Error('Body tag should exist!');
    }

    const toolbarItems = [
        {
            icon: 'fa-plus',
            type: 'button',
            onClick: jest.fn(),
        },
        {
            icon: 'fa-gear',
            type: 'dropdown',
            options: [
                {
                    label: 'Option1',
                    onClick: jest.fn(),
                },
            ],
        },
    ];

    const toolbar = mount(<Toolbar columnIndex={0} toolbarItems={toolbarItems} />);

    expect(toolbar.render()).toMatchSnapshot();
    expect(toolbar.find(ToolbarDropdown).length).toBe(1);

    toolbar.find('.fa-plus').simulate('click');
    expect(toolbarItems[0].onClick).toBeCalledWith(0);

    // check for opened dropdown in body
    expect(body.innerHTML).toBe('');
    toolbar.find(ToolbarDropdown).simulate('click');
    expect(pretty(body.innerHTML)).toMatchSnapshot();
});
