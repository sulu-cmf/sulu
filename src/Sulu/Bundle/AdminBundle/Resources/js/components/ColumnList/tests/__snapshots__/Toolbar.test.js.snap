// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Toolbar component should render with active 1`] = `
<div
  class="toolbar active"
>
  <div
    class="item primary"
  >
    <span
      aria-label="fa-plus"
      class="fa fa-plus"
    />
  </div>
  <div
    class="item primary"
  >
    <span
      aria-label="fa-search"
      class="fa fa-search"
    />
  </div>
  <div
    class="item primary"
  >
    <span
      aria-label="fa-gear"
      class="fa fa-gear"
    />
  </div>
</div>
`;

exports[`The Toolbar component should render with active 2`] = `"<div><div class=\\"backdrop fixed\\"></div></div><div><div class=\\"container\\"><div style=\\"top: 10px; position: fixed; pointer-events: auto; left: 10px;\\"><div class=\\"listContainer\\"><ul class=\\"list primary\\"><li><button class=\\"option\\">Option1</button></li><li><button class=\\"option\\">Option2</button></li></ul></div></div></div></div>"`;

exports[`The Toolbar component should render with active 3`] = `
<div
  class="toolbar"
>
  <div
    class="item primary"
  >
    <span
      aria-label="fa-plus"
      class="fa fa-plus"
    />
  </div>
  <div
    class="item primary"
  >
    <span
      aria-label="fa-search"
      class="fa fa-search"
    />
  </div>
  <div
    class="item primary"
  >
    <span
      aria-label="fa-gear"
      class="fa fa-gear"
    />
  </div>
</div>
`;
